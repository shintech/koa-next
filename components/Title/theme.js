export default ({ fontSize, colors }) => ({
  text: colors[0],
  textShadow: `0.08ch 0.08ch ${colors[1]}`,
  fontSize: fontSize
})
