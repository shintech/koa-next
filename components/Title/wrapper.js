import styled from 'styled-components'

const Wrapper = styled.div`
  h1 {
    font-family: 'Rouge Script', cursive;
    font-size: ${props => props.theme.fontSize};
    color: ${props => props.theme.text};
    text-shadow: ${props => props.theme.textShadow};
    margin: 0;
  }
`

export default Wrapper
