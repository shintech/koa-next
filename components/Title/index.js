import PropTypes from 'prop-types'
import Wrapper from './wrapper'
import theme from './theme'

const Title = ({ title = 'Hello World!', fontSize = '16ch', colors = ['orange', 'red'] }) =>
  <Wrapper theme={theme({ fontSize, colors })}>
    <h1 className='title'>{ title }</h1>
  </Wrapper>

Title.propTypes = {
  title: PropTypes.string.isRequired,
  fontSize: PropTypes.string,
  colors: PropTypes.array
}

export default Title
