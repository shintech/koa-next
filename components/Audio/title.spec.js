/* eslint-env jest */

import { shallow } from 'enzyme'
import { compose } from 'redux'
import toJSON from 'enzyme-to-json'
import Title from 'components/Title'

const shallowExpect = compose(expect, toJSON, shallow)

describe('COMPONENT -> Title...', () => {
  let clicker = shallow(<Title title='Hello World!' />)

  it(`expect Title h1 text to equal "title"...`, () => {
    expect(clicker.find('h1').text()).toBe('Hello World!')
  })
})

describe('COMPONENT -> Title - snapshot...', () => {
  it('expect to render correct properties', () => {
    shallowExpect(<Title title='Hello World!!' />).toMatchSnapshot()
  })
})
