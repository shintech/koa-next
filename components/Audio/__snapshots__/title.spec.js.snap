// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT -> Title - snapshot... expect to render correct properties 1`] = `
<wrapper__Wrapper
  theme={
    Object {
      "fontSize": "16ch",
      "text": "orange",
      "textShadow": "0.08ch 0.08ch red",
    }
  }
>
  <h1
    className="title"
  >
    Hello World!!
  </h1>
</wrapper__Wrapper>
`;
